lib/class-experimental-wp-widget-blocks-manager.php

```diff
@@ -271,4 +292,93 @@ public static function serialize_block( $block ) {
 			);
 		}
 	}
+
+	/**
+	 * Outputs a block widget on the website frontend.
+	 *
+	 * @param array $options   Widget options.
+	 * @param array $arguments Arguments array.
+	 */
+	public static function gutenberg_output_blocks_widget( $options, $arguments ) {
+		echo $options['before_widget'];
+		foreach ( $arguments['blocks'] as $block ) {
+			echo render_block( $block );
+		}
+		echo $options['after_widget'];
+	}
+
+	/**
+	 * Registers of a widget that should represent a set of blocks and returns its id.
+	 *
+	 * @param array $blocks   Array of blocks.
+	 */
+	public static function gutenberg_blocks_to_widget( $blocks ) {
+		$widget_id = 'blocks-widget-' . md5( Experimental_WP_Widget_Blocks_Manager::serialize_block( $blocks ) );
+		global $wp_registered_widgets;
+		if ( isset( $wp_registered_widgets[ $widget_id ] ) ) {
+			return $widget_id;
+		}
+		wp_register_sidebar_widget(
+			$widget_id,
+			__( 'Blocks Area ', 'gutenberg' ),
+			'Experimental_WP_Widget_Blocks_Manager::gutenberg_output_blocks_widget',
+			array(
+				'classname'   => 'widget-area',
+				'description' => __( 'Displays a set of blocks', 'gutenberg' ),
+			),
+			array(
+				'blocks' => $blocks,
+			)
+		);
+		return $widget_id;
+	}
+
+	/**
+	 * Filters the $sidebars_widgets to exchange wp_area post id with a widget that renders that block area.
+	 *
+	 * @param array $sidebars_widgets_input An associative array of sidebars and their widgets.
+	 */
+	public static function gutenberg_swap_out_sidebars_blocks_for_block_widgets( $sidebars_widgets_input ) {
+		global $sidebars_widgets;
+		if ( null === self::$unfiltered_sidebar_widgets ) {
```

Comment:
If `gutenberg_swap_out_sidebars_blocks_for_block_widgets` is called again with a different `$sidebars_widgets_input` then this won't update its cached copy of `self::$unfiltered_sidebar_widgets`. 

We probably want:

```php
if ( $sidebars_widgets_input !== self::$unfiltered_sidebar_widgets ) {
```

Reply:
Actually, not sure about this. Maybe it's not a problem. Is `global $sidebars_widgets` ever updated?